paths_cli/tests/compiling/utils.py

```diff
@@ -0,0 +1,9 @@
+from paths_cli.compiling.core import CategoryCompiler
+
+def mock_compiler(category, type_dispatch=None, named_objs=None):
+    if type_dispatch is None:
+        type_dispatch = {}
```

Comment:
can be shortened
```suggestion
    type_dispatch = type_dispatch or {}
```

Reply:
There's a slight difference in behavior -- not one that actually matters, I don't think, but my intuition would expect the behavior of the current implementation, so keeping this as-is.

```python
td = {}
c = mock_compiler('category', type_dispatch=td)
... # register 'some_key' with type_dispatch
assert 'some_key' in c.type_dispatch  # True
assert 'some_key' in td  # depends on implementation
```

However, the `or` is a nice trick that I should probably be using in many cases that I created the utility function `none_to_default` for (used widely in the SimStore stuff.)

Reply:
> However, the or is a nice trick that I should probably be using in many cases that I created the utility function none_to_default for (used widely in the SimStore stuff.)


This trick only works for input that isn't allowed to be `falsey` ;) So if empty lists/sets/dicts/strings, False, 0 are valid inputs that should not be overwritten then this would not work. (As shown by your example)
 
Another common failure case for this trick is having a `None` default, that is replaced by some positive int but also having the possibility of `0`  (i.e. buffersize where 0 is unbuffered, but default None is transformed to  `n` bytes)